tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -900,6 +909,42 @@ Status FragmentMetadata::load_rtree(const EncryptionKey& encryption_key) {
   return Status::Ok();
 }
 
+void FragmentMetadata::free_rtree() {
+  auto freed = rtree_.free_memory();
+  memory_tracker_->release_memory(freed);
+  loaded_metadata_.rtree_ = false;
+}
+
+void FragmentMetadata::free_tile_offsets() {
+  for (auto v : tile_offsets_) {
+    memory_tracker_->release_memory(v.size() * sizeof(uint64_t));
+    v.clear();
+  }
+  loaded_metadata_.tile_offsets_.resize(
+      array_schema_->attribute_num() + 1, false);
+
+  for (auto v : tile_var_offsets_) {
+    memory_tracker_->release_memory(v.size() * sizeof(uint64_t));
+    v.clear();
+  }
+  loaded_metadata_.tile_var_offsets_.resize(
+      array_schema_->attribute_num() + 1, false);
+
+  for (auto v : tile_var_sizes_) {
+    memory_tracker_->release_memory(v.size() * sizeof(uint64_t));
+    v.clear();
+  }
+  loaded_metadata_.tile_var_sizes_.resize(
+      array_schema_->attribute_num() + 1, false);
+
+  for (auto v : tile_validity_offsets_) {
+    memory_tracker_->release_memory(v.size() * sizeof(uint64_t));
+    v.clear();
+  }
+  loaded_metadata_.tile_validity_offsets_.resize(
+      array_schema_->attribute_num() + 1, false);
```

Comment:
Are the .resize()s used in this routine expected to (re-)set the vector elements to false?
If so, I don't think that will happen, think only added elements will be assigned the 'init' value, others will be left unchanged.

Reply:
Also noticed this routine uses .resize(... ->attribute_num() + 1...) 
while a number of other (pre-existing?) places in file for same items 
seem to just use .resize(... ->attribute_num() /*without +1 */ ... )
Not sure if it would matter, just noticed...